From 38dc70ee7a8f3b25748eab0f6bb98422fa76f79d Mon Sep 17 00:00:00 2001
From: Didier 'OdyX' Raboud <didier.raboud@liip.ch>
Date: Mon, 5 Feb 2018 17:13:27 +0100
Subject: Set the paths according to where the files are shipped

 * Code goes to /usr/share/php/passbolt-api/
 * Configuration goes to /etc/passbolt/
 * Variable files (cold cache and GPG keys) go to /var/lib/passbolt/
 * Logs go to /var/log/passbolt/
---
 bin/cake.php      |  4 ++--
 config/paths.php  | 20 +++++++++++++++-----
 webroot/index.php |  4 ++--
 3 files changed, 19 insertions(+), 9 deletions(-)

diff --git a/bin/cake.php b/bin/cake.php
index 68cd5b4db2..eb6cfac61c 100644
--- a/bin/cake.php
+++ b/bin/cake.php
@@ -1,10 +1,10 @@
 #!/usr/bin/php -q
 <?php
 // Check platform requirements
-require dirname(__DIR__) . '/config/requirements.php';
+require '/etc/passbolt/requirements.php';
 require dirname(__DIR__) . '/vendor/autoload.php';
 use App\Application;
 use Cake\Console\CommandRunner;
 // Build the runner with an application and root executable name.
-$runner = new CommandRunner(new Application(dirname(__DIR__) . '/config'), 'cake');
+$runner = new CommandRunner(new Application('/etc/passbolt'), 'cake');
 exit($runner->run($argv));
\ No newline at end of file
diff --git a/config/paths.php b/config/paths.php
index 95a7e0f286..31f961957a 100644
--- a/config/paths.php
+++ b/config/paths.php
@@ -28,7 +28,17 @@ if (!defined('DS')) {
 /**
  * The full path to the directory which holds "src", WITHOUT a trailing DS.
  */
-define('ROOT', dirname(__DIR__));
+define('ROOT', '/usr/share/php/passbolt-api');
+
+/**
+ * The full path to the "variable data" passbolt working directory, WITHOUT a trailing DS.
+ */
+define('VAR_DIR', '/var/lib/passbolt');
+
+/**
+ * The full path to the "logs" passbolt working directory, WITHOUT a trailing DS.
+ */
+define('LOG_DIR', '/var/log/passbolt');
 
 /**
  * The actual directory name for the application directory. Normally
@@ -44,7 +54,7 @@ define('APP', ROOT . DS . APP_DIR . DS);
 /**
  * Path to the config directory.
  */
-define('CONFIG', ROOT . DS . 'config' . DS);
+define('CONFIG', '/etc/passbolt' . DS);
 
 /**
  * File path to the webroot directory.
@@ -54,7 +64,7 @@ define('WWW_ROOT', ROOT . DS . 'webroot' . DS);
 /**
  * File path to the images directory.
  */
-define('IMAGES', WWW_ROOT . 'img' . DS);
+define('IMAGES',  VAR_DIR . DS . 'img' . DS);
 
 /**
  * Path to the tests directory.
@@ -69,12 +79,12 @@ define('FIXTURES', TESTS . 'Fixture' . DS);
 /**
  * Path to the temporary files directory.
  */
-define('TMP', ROOT . DS . 'tmp' . DS);
+define('TMP', VAR_DIR . DS . 'tmp' . DS);
 
 /**
  * Path to the logs directory.
  */
-define('LOGS', ROOT . DS . 'logs' . DS);
+define('LOGS', LOG_DIR . DS);
 
 /**
  * Path to the cache files directory. It can be shared between hosts in a multi-server setup.
diff --git a/webroot/index.php b/webroot/index.php
index 3c895caf12..16622914c2 100644
--- a/webroot/index.php
+++ b/webroot/index.php
@@ -14,7 +14,7 @@
  * @since         0.2.9
  * @license       MIT License (http://www.opensource.org/licenses/mit-license.php)
  */
-require dirname(__DIR__) . '/config/requirements.php';
+require '/etc/passbolt/requirements.php';
 
 // for built-in server
 if (php_sapi_name() === 'cli-server') {
@@ -32,7 +32,7 @@ use App\Application;
 use Cake\Http\Server;
 
 // Bind your application to the server.
-$server = new Server(new Application(dirname(__DIR__) . '/config'));
+$server = new Server(new Application('/etc/passbolt'));
 
 // Run the request/response through the application
 // and emit the response.
